internal/controllers/user_devices_controller.go

```diff
@@ -231,75 +231,80 @@ func (udc *UserDevicesController) dbDevicesToDisplay(ctx context.Context, device
 		var nft *VehicleNFTData
 		var credential *VINCredentialData
 		pu := []PrivilegeUser{}
-		if !d.TokenID.IsZero() {
-			nftStatus := d.R.MintRequest
-			nft = &VehicleNFTData{
-				Status: nftStatus.Status,
-			}
-			if nftStatus.Hash.Valid {
-				hash := hexutil.Encode(nftStatus.Hash.Bytes)
-				nft.TxHash = &hash
-			}
 
-			nft.TokenID = d.TokenID.Int(nil)
-			nft.TokenURI = fmt.Sprintf("%s/v1/vehicle/%s", udc.Settings.DeploymentBaseURL, nft.TokenID)
-			if d.OwnerAddress.Valid {
+		if !d.TokenID.IsZero() || d.R.MintRequest != nil {
+			nft = &VehicleNFTData{}
+
+			if !d.TokenID.IsZero() {
+				nft.TokenID = d.TokenID.Int(nil)
+
+				nft.TokenURI = fmt.Sprintf("%s/v1/vehicle/%s", udc.Settings.DeploymentBaseURL, nft.TokenID)
+
 				addr := common.BytesToAddress(d.OwnerAddress.Bytes)
 				nft.OwnerAddress = &addr
-			}
 
-			// NFT Privileges
-			udp, err := models.NFTPrivileges(
-				models.NFTPrivilegeWhere.TokenID.EQ(types.Decimal(d.TokenID)),
-				models.NFTPrivilegeWhere.Expiry.GT(time.Now()),
-				models.NFTPrivilegeWhere.ContractAddress.EQ(common.FromHex(udc.Settings.VehicleNFTAddress)),
-			).All(ctx, udc.DBS().Reader)
-			if err != nil {
-				return nil, err
-			}
+				// NFT Privileges
+				udp, err := models.NFTPrivileges(
+					models.NFTPrivilegeWhere.TokenID.EQ(types.Decimal(d.TokenID)),
```

Comment:
This was actually a bug (well, an inefficiency, we would search for token id 0 for in-progress mints I think) in the old code.